builder/vsphere/driver/vm.go

```diff
@@ -205,8 +205,11 @@ func (d *Driver) CreateVM(config *CreateConfig) (*VirtualMachine, error) {
 		return nil, err
 	}
 
-	vmRef := taskInfo.Result.(types.ManagedObjectReference)
+	if taskInfo.Result == nil {
+		return nil, fmt.Errorf("something went wrong when creating the VM")
+	}
 
+	vmRef := taskInfo.Result.(types.ManagedObjectReference)
```

Comment:
Nice, you can also use a type assertion ok check to verify that taskInfo.Result is what we expect it to be. If taskInfo.Result is nil `ok` will be false. If it is not a `types.ManagedObjectReference` it will also be false. 
```suggestion
	vmRef, ok := taskInfo.Result.(types.ManagedObjectReference)
	if !ok {
		return nil, fmt.Errorf("something went wrong when creating the VM")
	}
```

Reply:
oh ! I completely forgot that we are to do that 🤦  thank you!

Reply:
You got it! It happens to me as well. I was just working with a similar issue so it was top of mind :+1: